.github/workflows/ci.yml

```diff
@@ -6,51 +6,33 @@ on:
       - master
   pull_request:
 
-jobs:
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - run: |
-          rustup override set stable
-          rustup update stable
-      - uses: Swatinem/rust-cache@v1.3.0
-      - run: cargo check
-        env:
-          RUSTFLAGS: "-D warnings"
+env:
+  # minimum supported rust version
+  MSRV: 1.51.0
 
+jobs:
   tests:
     name: Tests
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: |
-          rustup override set stable
-          rustup update stable
+          rustup override set ${{ env.MSRV }}
+          rustup update ${{ env.MSRV }}
       - uses: Swatinem/rust-cache@v1.3.0
       - run: cargo test
+        env:
+          RUSTFLAGS: "-D warnings"
 
-  fmt:
-    name: Rustfmt
+  lints:
+    name: Rustfmt and Clippy
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: |
-          rustup override set stable
-          rustup update stable
-      - run: rustup component add rustfmt
+          rustup override set ${{ env.MSRV }}
+          rustup update ${{ env.MSRV }}
```

Comment:
the `rustup update` calls are no longer necessary when we use specific version numbers, right?

Reply:
Good point! Yes, we could remove it as caching won't trick anymore :)